R/beers-r6.R

```diff
@@ -0,0 +1,153 @@
+Beers_Interpolator <- R6::R6Class("Beers_Interpolator",
+
+  public = list(
+    b1 = NULL,
+    b2 = NULL,
+    bm = NULL,
+    bp = NULL,
+    bf = NULL,
+
+    initialize = function(b1, b2, bm, bp, bf) {
+      self$b1 <- b1
+      self$b2 <- b2
+      self$bm <- bm
+      self$bp <- bp
+      self$bf <- bf
+    },
+
+    interpolate = function(points) {
+      if (length(points) < 6) {
+        stop("Beers interpolation requires at least 6 points")
+
+      } else if (!is.numeric(points)) {
+        stop("Beers interpolation requires numeric argument")
+      }
+
+      res <- rep(0, ((length(points) * 5) - 4))
+      for (p in 1:(length(points) - 1)) {
```

Comment:
Why a for loop followed by a bunch of exact tests against the index?  that warrants some explanation

Reply:
Simplified. The original for loop included every interpolation, and then the "if" statements handled the 4 special cases at the beginning and end. But yes, much simpler to do the special cases outside the loop...